app/models/student.rb

```diff
@@ -45,13 +45,25 @@ def activate!
 
   private
 
-  def update_volunteer_assignment(volunteer_id)
-    if active_student_volunteer_assignment.present? &&
-        active_student_volunteer_assignment.volunteer_id.to_s != volunteer_id
-      active_student_volunteer_assignment.update!(end_date: Date.current)
-      StudentVolunteerAssignment.create!(student: self, volunteer_id: volunteer_id, start_date: Date.current) if volunteer_id.present?
-    elsif active_student_volunteer_assignment.blank?
-      StudentVolunteerAssignment.create!(student: self, volunteer_id: volunteer_id, start_date: Date.current) if volunteer_id.present?
+  def update_volunteer_assignments(volunteer_ids: [])
+    active_volunteer_ids = active_student_volunteer_assignments.pluck(:volunteer_id).map(&:to_s).sort || []
+    volunteer_ids ||= []
+
+    return if active_volunteer_ids == volunteer_ids&.sort
+    volunteer_ids.compact_blank!
+
+    inactive_ids = active_volunteer_ids - volunteer_ids
+    active_student_volunteer_assignments.where(volunteer_id: inactive_ids).update(end_date: Date.current) if inactive_ids.present?
+
+    new_volunteers = volunteer_ids - active_volunteer_ids
+    if new_volunteers.present?
+      StudentVolunteerAssignment
+        .insert_all!(new_volunteers.map { |volunteer_id|
+                       {student_id: id,
+                        volunteer_id: volunteer_id,
+                        created_at: Time.zone.now,
+                        updated_at: Time.zone.now}
+                     })
```

Comment:
Interesting! I wasn't familiar with `insert_all`. Looks like it avoids callbacks and manually requires setting timestamps. Let's not use this here.

At CEP, most students only have one volunteer. But there are a few with 2 or 3. So I don't think we'd hit a performance issue that requires us to need this speedier update, and sacrifice validations/callbacks.

Reply:
ack. Thank you for sharing the info. I have done the update. Please check.